src/runtime_src/core/common/scheduler.cpp

```diff
@@ -184,8 +184,7 @@ init(xclDeviceHandle handle, const axlf* top)
       scmd->count = sizeof (ert_configure_sk_cmd) / 4 - 1;
       scmd->start_cuidx = start_cuidx;
       scmd->num_cus = sk.ninst;
-      std::strncpy(reinterpret_cast<char*>(scmd->sk_name), sk.symbol_name, 31);
-      reinterpret_cast<char*>(scmd->sk_name)[31] = 0;
+      sk.symbol_name.copy(reinterpret_cast<char*>(scmd->sk_name), 31);
```

Comment:
is null termination of destination string guaranteed?
I think Coverity will complain here.

Reply:
earlier clear of scmd by "std::memset(scmd, 0, 0x1000);" ensures that this string will be null terminated. copy is only for max of 31 characters. So 32nd char will be always null. I will check coverity output to see what it says.

Reply:
Yep. I missed the memset on line 180.